### PR TITLE
Fix issues with REWRITE_DONE in floating point rewriter

### PR DESCRIPTION
**Pull request by ajreynol**
This was triggered by sygus + FP with fp.min/fp.max in the grammar.

### CODE REVIEW

File: src/theory/fp/theory_fp_rewriter.cpp [hunk 1]
```diff
@@ -175,8 +175,9 @@ namespace rewrite {
     if (node[0] == node[1]) {
       return RewriteResponse(REWRITE_DONE, NodeManager::currentNM()->mkConst(true));
     } else if (!isPreRewrite && (node[0] > node[1])) {
-	Node normal = NodeManager::currentNM()->mkNode(kind::EQUAL,node[1],node[0]);
-	return RewriteResponse(REWRITE_DONE, normal);
+      Node normal =
+          NodeManager::currentNM()->mkNode(kind::EQUAL, node[1], node[0]);
+      return RewriteResponse(REWRITE_AGAIN_FULL, normal);
```
Comment (4tXJ7f) [inline src/theory/fp/theory_fp_rewriter.cpp]: This change should not be needed (there is no other rewrite that could apply and `node[0]`/`node[1]` are already rewritten at that point.

File: src/theory/fp/theory_fp_rewriter.cpp [hunk 2]
```diff
@@ -191,7 +192,7 @@ namespace rewrite {
 	   (k == kind::FLOATINGPOINT_MIN_TOTAL) || (k == kind::FLOATINGPOINT_MAX_TOTAL));
 #endif
     if (node[0] == node[1]) {
-      return RewriteResponse(REWRITE_DONE, node[0]);
+      return RewriteResponse(REWRITE_AGAIN_FULL, node[0]);
```
Comment (4tXJ7f) [inline src/theory/fp/theory_fp_rewriter.cpp]: `REWRITE_AGAIN` should be sufficient here (to make sure that we rewrite as much as possible on the prerewriting pass).

File: src/theory/fp/theory_fp_rewriter.cpp [hunk 3]
```diff
@@ -204,7 +205,7 @@ namespace rewrite {
 
     if (node[0] > node[1]) {
       Node normal = NodeManager::currentNM()->mkNode(kind::FLOATINGPOINT_EQ,node[1],node[0]);
-      return RewriteResponse(REWRITE_DONE, normal);
+      return RewriteResponse(REWRITE_AGAIN_FULL, normal);
```
Comment (4tXJ7f) [inline src/theory/fp/theory_fp_rewriter.cpp]: This change should not be needed (there is no other rewrite that could apply and `node[0]`/`node[1]` are already rewritten (or just about to be rewritten) at that point.

File: src/theory/fp/theory_fp_rewriter.cpp [hunk 4]
```diff
@@ -217,7 +218,7 @@ namespace rewrite {
 
     if (node[1] > node[2]) {
       Node normal = NodeManager::currentNM()->mkNode(k,node[0],node[2],node[1]);
-      return RewriteResponse(REWRITE_DONE, normal);
+      return RewriteResponse(REWRITE_AGAIN_FULL, normal);
```
Comment (4tXJ7f) [inline src/theory/fp/theory_fp_rewriter.cpp]: Ditto.

File: src/theory/fp/theory_fp_rewriter.cpp [hunk 5]
```diff
@@ -229,7 +230,7 @@ namespace rewrite {
 
     if (node[1] > node[2]) {
       Node normal = NodeManager::currentNM()->mkNode(kind::FLOATINGPOINT_FMA,node[0],node[2],node[1],node[3]);
-      return RewriteResponse(REWRITE_DONE, normal);
+      return RewriteResponse(REWRITE_AGAIN_FULL, normal);
```
Comment (4tXJ7f) [inline src/theory/fp/theory_fp_rewriter.cpp]: Ditto.

File: src/theory/fp/theory_fp_rewriter.cpp [hunk 6]
```diff
@@ -280,7 +281,11 @@ namespace rewrite {
 			   nm->mkNode(kind::FLOATINGPOINT_REM, working[0][0], working[1]));
     }
 
-    return RewriteResponse(REWRITE_DONE, working);
+    if (working != node)
+    {
+      return RewriteResponse(REWRITE_AGAIN_FULL, working);
```
Comment (4tXJ7f) [inline src/theory/fp/theory_fp_rewriter.cpp]: Technically, not all rewrites need `REWRITE_AGAIN_FULL`, only the last one.

File: src/theory/fp/theory_fp_rewriter.cpp (hunk 1 above)
Comment (ajreynol) [reply src/theory/fp/theory_fp_rewriter.cpp]: Yes, you're right.

File: src/theory/fp/theory_fp_rewriter.cpp (hunk 3 above)
Comment (ajreynol) [reply src/theory/fp/theory_fp_rewriter.cpp]: Right.

**Pull request merged**